libr/core/cmd_search.c

```diff
@@ -500,38 +486,35 @@ static int __cb_hit(RSearchKeyword *kw, void *user, ut64 addr) {
 			}
 		} else {
 			r_cons_printf ("0x%08"PFMT64x " %s%d_%d %s\n",
-				base_addr + addr, searchprefix, kw->kwidx, kw->count, s);
+				base_addr + addr, searchprefix, kw->kwidx, kw->count - 1, s);
 		}
 		free (s);
 		free (buf);
 		free (str);
 	} else if (kw) {
 		if (json) {
-			if (!first_hit) {
+			if (core->search->nhits > 1) {
 				r_cons_printf (",");
 			}
 			r_cons_printf ("{\"offset\": %"PFMT64d ",\"len\":%d}",
 				base_addr + addr, kw->kwidx, kw->keyword_length);
 		} else {
 			if (searchflags) {
-				r_cons_printf ("%s%d_%d\n", searchprefix, kw->kwidx, kw->count);
+				r_cons_printf ("%s%d_%d\n", searchprefix, kw->kwidx, kw->count - 1);
 			} else {
 				r_cons_printf ("f %s%d_%d %d 0x%08"PFMT64x "\n", searchprefix,
-					kw->kwidx, kw->count, kw->keyword_length, base_addr + addr);
+					kw->kwidx, kw->count - 1, kw->keyword_length, base_addr + addr);
 			}
 		}
 	}
-	if (first_hit) {
-		first_hit = false;
-	}
 	if (searchflags && kw) {
-		const char *flag = sdb_fmt (0, "%s%d_%d", searchprefix, kw->kwidx, kw->count);
+		const char *flag = sdb_fmt (0, "%s%d_%d", searchprefix, kw->kwidx, kw->count - 1);
 		r_flag_set (core->flags, flag, base_addr + addr, kw->keyword_length);
 	}
-	if (!STRNULL (cmdhit)) {
+	if (*param->cmd_hit) {
```

Comment:
r_config_get can return NULL so dereferencing cmd_hit here can cause segfault

Reply:
`cmd.hit` is defined in `cconfig.c` so `r_config_get` cannot return `NULL`.